### DISCUSSION TITLE
数据库类型支持OceanBase讨论

### DISCUSSION THREAD

**Discussion by ghost** [hhyo/Archery#2562] [Ideas]
因为工作需要，2023年引入了Archery产品作为数据库管理的底座，并成功落地Oracle数据库自动化变更应用。由于公司的数据库发展，Oracle要换成OceanBase，我计算2024年对OceanBase进行Engine开发，后续也会将OceanBase对应的Engine分享给Archery。

由于[Python 应用程序连接 OceanBase 数据库](https://www.oceanbase.com/docs/enterprise-oceanbase-database-cn-10000000000362782)有前提条件，必须确保计算机上的 Java 环境为 Java JDK 8 版本。

将导致基础镜像变得特别大，有啥更好的办法解决呢

**Comment by LeoQuote** [comment]
这个比较难做，看看有没有兼容协议凑合下的，或者说把archery 核心抽象成 python 库，然后你engine 也是一个库，这样由用户自由组装

**Reply by LeoQuote** [reply]
https://www.oceanbase.com/docs/community-observer-cn-0000000000540981 就官方文档的话，是可以兼容 mysql协议的，可以考虑用这种方式接入

**Comment by LeoQuote** [comment]
或者说如果觉得可以，用微服务的方式远程web调用，mTLS 的情况下，应该也可以接受

**Reply by ghost** [reply]
谢谢建议，我去调研尝试。后续有进展我再回复情况。

**Comment by dbaops-com** [comment]
集成Oceanbase这个有进展么，我们最近也需要
